### PR TITLE
Changes allowing vertx to replace webgui

### PR DESCRIPTION
**Pull request by supertick**

### CODE REVIEW

File: src/main/java/org/myrobotlab/service/Vertx.java [hunk 1]
```diff
@@ -184,7 +196,34 @@ public boolean isLocal(Message msg) {
...
+   * 
+   * @param port
+   */
+  public void setPort(int port) {
+    config.port = port;
+    boolean wasListening = isListening;
+    if (isListening) {
+      stop();
+    }
+    sleep(2000);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Vertx.java:211]: do we really need this sleep here?  shouldn't stop block until it's actually stopped?

File: src/main/java/org/myrobotlab/service/Vertx.java [hunk 2]
```diff
@@ -41,6 +43,11 @@ public class Vertx extends Service<VertxConfig> implements Gateway {
 
   private transient io.vertx.core.Vertx vertx = null;
 
+  /**
+   * If listening currently on port
+   */
+  protected boolean isListening = false;
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/service/Vertx.java]: we should rename this to just "listening"

File: src/main/java/org/myrobotlab/vertx/ApiHandler.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+      // content type always json for now
+      request.response().putHeader("content-type", "application/json");
+
+      // POST vs GET handler ?
+      if ("get".equals(verb)) {
+        handleGet(rc);
+      } else if ("post".equals(verb)) {
+        handlePost(rc);
+      } else {
+        log.info("got a request to handle http verb %s", verb);
```
Comment (kwatters) [inline src/main/java/org/myrobotlab/vertx/ApiHandler.java]: this should probably be a warn level.

File: src/main/java/org/myrobotlab/service/Vertx.java (hunk 2 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/service/Vertx.java]: updated

File: src/main/java/org/myrobotlab/vertx/ApiHandler.java (hunk 3 above)
Comment (supertick) [reply src/main/java/org/myrobotlab/vertx/ApiHandler.java]: updated

**Pull request merged**